packages/tendermint-rpc/src/rpcclients/httpbatchclient.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  isJsonRpcErrorResponse,
+  JsonRpcRequest,
+  JsonRpcSuccessResponse,
+  parseJsonRpcResponse,
+} from "@cosmjs/json-rpc";
+
+import { http, HttpEndpoint } from "./httpclient";
+import { hasProtocol, RpcClient } from "./rpcclient";
+
+export interface HttpBatchClientOptions {
+  dispatchInterval: number;
+  batchSizeLimit: number;
+}
+
+export const defaultHttpBatchClientOptions: HttpBatchClientOptions = {
+  dispatchInterval: 20,
+  batchSizeLimit: 20,
+};
+
+export class HttpBatchClient implements RpcClient {
+  protected readonly url: string;
+  protected readonly headers: Record<string, string> | undefined;
+  protected readonly options: HttpBatchClientOptions;
+  private timer?: NodeJS.Timer;
+
+  private readonly queue: Array<{
+    request: JsonRpcRequest;
+    resolve: (a: JsonRpcSuccessResponse) => void;
+    reject: (a: Error) => void;
+  }> = [];
+
+  public constructor(
+    endpoint: string | HttpEndpoint,
+    options: HttpBatchClientOptions = defaultHttpBatchClientOptions,
+  ) {
+    this.options = options;
+    if (typeof endpoint === "string") {
+      // accept host.name:port and assume http protocol
+      this.url = hasProtocol(endpoint) ? endpoint : "http://" + endpoint;
+    } else {
+      this.url = endpoint.url;
+      this.headers = endpoint.headers;
+    }
+    this.timer = setInterval(() => this.tick(), options.dispatchInterval);
+    this.validate();
+  }
+
+  public disconnect(): void {
+    this.timer && clearInterval(this.timer);
+    this.timer = undefined;
+  }
+
+  public async execute(request: JsonRpcRequest): Promise<JsonRpcSuccessResponse> {
+    return new Promise((resolve, reject) => {
+      this.queue.push({ request, resolve, reject });
+    });
+  }
+
+  private validate(): void {
+    if (this.options.batchSizeLimit < 1) throw new Error("batchSizeLimit < 1");
```

Comment:
Let's check for truthiness first. There are those falsy values of `number`, like `NaN`.

```
> NaN < 1
false
```